### PR TITLE
improved touch and backlight handling

### PR DESCRIPTION
**Pull request by gagarinlg**
This improves the rouch enable/disable handling and backlight<->touch dependencies.
This PR need an upcoming PR for edgtetx to work

### CODE REVIEW

**Comment by raphaelcoeffic** [comment]
I will need a bit to understand, as I’m surprised at first of the number of states involved. @pfeerick can you please have a look as well?

**Comment by gagarinlg** [comment]
I was also, but the problem here is, that you have to take care of the touch state when enabling/disabling touch handling. I think that it is bad practice and can lead to undefined behavior, if you do switch touch handling of and the last event was a touch down/slide event, because the end of the touch is then missing.

**Comment by pfeerick** [comment]
Whilst at first I was also surprised, it makes sense, as there is the initial touch, and then still being touched, and then has just been released. More importantly, I think it is trying to make crystal clear what is when touch is enabled or disabled, and how to handle the initial touch event, touch held, and touch released, for both enable and disable states. I'm sure it will become clearer once I have a proper look at the code in effect ;)

**Comment by gagarinlg** [comment]
> More importantly, I think it is trying to make crystal clear what is when touch is enabled or disabled, and how to handle the initial touch event, touch held, and touch released, for both enable and disable states. 

This was exactly my intention, make clear what is happening.

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -39,15 +39,84 @@ void MainWindow::emptyTrash()
...
+}
+#endif
+
 void MainWindow::checkEvents()
 {
 #if defined(HARDWARE_TOUCH)
+
   if (touchPanelEventOccured()) {
-    touchPanelRead();
+    touchState = touchPanelRead();
```
Comment (pfeerick) [inline src/mainwindow.cpp:80]: touchPanelRead is void function, so this doesn't work ;)

**Review by pfeerick — CHANGES_REQUESTED** [review]

File: src/mainwindow.cpp (hunk 1 above)
Comment (gagarinlg) [reply src/mainwindow.cpp:80]: It is not anymore, with the changes from the PR #1000 on EdgeTX. So they depend on each other.

Comment (pfeerick) [reply src/mainwindow.cpp:80]: Thanks, sorry about that!

**Pull request merged**